auto_round/export/export_to_gguf/export.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import shutil
+import torch
+from .convert import Model
+from auto_round.utils import logger
+from pathlib import Path
+
+import gguf
+
+
+FTYPE_MAP: dict[str, gguf.LlamaFileType] = {
+        "f32": gguf.LlamaFileType.ALL_F32,
+        "f16": gguf.LlamaFileType.MOSTLY_F16,
+        "bf16": gguf.LlamaFileType.MOSTLY_BF16,
+        "q8_0": gguf.LlamaFileType.MOSTLY_Q8_0,
+        "q4_0": gguf.LlamaFileType.MOSTLY_Q4_0,
+        "q4_1": gguf.LlamaFileType.MOSTLY_Q4_1,
+        "q4_k": gguf.LlamaFileType.MOSTLY_Q4_K_S,
+        "auto": gguf.LlamaFileType.GUESSED,
+    }
+
+def save_quantized_as_gguf(output_dir, backend="gguf:q4_0", **kwargs):
+    """Export the model to gguf format."""
+
+    model = kwargs["model"]
+    tokenizer = kwargs.get("tokenizer", None)
+    config = model.config
+
+    tmp_work_dir = Path(os.path.join(output_dir, 'tmp_dir'))
+    if tokenizer is not None:
+        tokenizer.save_pretrained(tmp_work_dir)
+    config.save_pretrained(tmp_work_dir)
+
+    with torch.inference_mode():
```

Comment:
have a test of phi-2 and phi-3, , as I recall there were some issues with this context previously.

Reply:
Test on phi2, phi-3-mini-4k-instruct. Work well.